src/steamcompmgr.cpp

```diff
@@ -7821,7 +7844,7 @@ steamcompmgr_main(int argc, char **argv)
 
 		// If we are running behind, allow tearing.
 
-		const bool bForceRepaint = g_bForceRepaint.exchange(false);
+		const bool bForceRepaint = vblank && g_bForceRepaint.exchange(false);
```

Comment:
This wasn't working without the `vblank` check due to the logic on line 7891:
```
bShouldPaint = vblank && ( hasRepaint || hasRepaintNonBasePlane || bForceSyncFlip );
```

Note that this will ignore our `bForceRepaint` request (that gets passed via `bForceSyncFlip`) if `vblank` is false. But by the time this happens, our originally force repaint request is lost because `g_bForceRepaint.exchange(false)` has already been called. So any time this line would get hit while `vblank` is false will result in the request getting lost. My update makes it so we don't reset the `g_bForceRepaint` request unless it will actually trigger a repaint.

Reply:
Yeah, makes sense for force repaint to check for vblank.